### DISCUSSION TITLE
Adding up accumulations in the same file using pcp_combine

### DISCUSSION THREAD

**Discussion by mpm-meto** [dtcenter/METplus#2233] [Configuration] [answered]
We think we can get pcp_combine to work on the command line using MET only to add up 6h lead times **in the same file** to create a 24h accumulation forecast.
i.e. pcp_combine -add 20210324T0000Z_um_diag.nc 'name="precipitation_amount"; level="(0,*,*)";' 20210324T0000Z_um_diag.nc 'name="precipitation_amount"; level="(1,*,*)";' 20210324T0000Z_um_diag.nc 'name="precipitation_amount"; level="(2,*,*)";' 20210324T0000Z_um_diag.nc 'name="precipitation_amount"; level="(3,*,*)";' sample_fcst24.nc
This takes 6h,12h,18h and 24h 6h accums, all in the same file, and adds them together. 

**So far we can't get the same to work using a metplus wrapper.** The best we have managed is to get it to add up 4 accums but it's always index 0. The command it says it's trying to run is: 
pcp_combine -v 5 -add 20210324T0000Z_um_diag.nc 'name="precipitation_amount"; level="(0S,*,*)";' 20210324T0000Z_um_diag.nc 'name="precipitation_amount"; level="(0S,*,*)";' 20210324T0000Z_um_diag.nc 'name="precipitation_amount"; level="(0S,*,*)";' 20210324T0000Z_um_diag.nc 'name="precipitation_amount"; level="(0S,*,*)";' -name "precipitation_amount" 20210324T0000Z_um_diag_f024_A24.nc
This fails with:
ERROR  : 
ERROR  : VarInfoNcCF::set_magic() -> trouble parsing NetCDF dimension value "0S"!
ERROR  : 

This is based on:
VALID_TIME_FMT = %Y%m%d%H
VALID_BEG = 2021032500
VALID_END = 2021032500
VALID_INCREMENT = 6H
LEAD_SEQ = 6
LOOP_ORDER = times
FCST_PCP_COMBINE_RUN = True
FCST_PCP_COMBINE_METHOD = ADD
FCST_PCP_COMBINE_INPUT_NAMES = [precipitation_amount,precipitation_amount,precipitation_amount,precipitation_amount]
FCST_PCP_COMBINE_INPUT_ACCUMS = 6H,6H,6H,6H
FCST_PCP_COMBINE_INPUT_LEVELS = "({lead?shift=-21600},*,*)","({lead?shift=-43200},*,*)","({lead?offset=-64800},*,*)","({lead?shift=0},*,*)"

We're  not really sure whether to use init or valid TBH but the outcome in terms of using the shift command string in the levels is the same. It *looks* like this hasn't been implemented in the wrapper. Could you confirm please? i.e. the accums MUST be in separate files.

**Comment by georgemccabe** [comment]
Hi @mpm-meto, could you share your METplus configuration file and the input file that you are using?

**Reply by georgemccabe** [reply]
Hi @mpm-meto, I am just following up on this discussion. Were you able to resolve the issue or do you need more assistance? If the latter, could you please send your configuration and input file? Thanks.

**Reply by georgemccabe** [reply]
Hi @mpm-meto, just following up again. If I don't hear from you in a week I will close this discussion. Thanks!

**Comment by georgemccabe** [comment, marked as answer]
Without an input file and full configuration file it is difficult to test if PCPCombine wrapper can be configured to build this command using the ADD method. My guess is that it is not possible to accomplish this using `FCST_PCP_COMBINE_METHOD = ADD` without enhancements to the wrapper. There are many ways to call pcp_combine and the ADD method in the wrapper was designed with a specific set of use cases in mind.

However, you should be able to use the USER_DEFINED method to build the command. This feature allows you to explicitly specify the arguments to pcp_combine (except for the output file name, which is still controlled by `FCST_PCP_COMBINE_OUTPUT_TEMPLATE`).

Here is an example use case for PCPCombine User Defined method: https://metplus.readthedocs.io/en/latest/generated/met_tool_wrapper/PCPCombine/PCPCombine_user_defined.html

Here is an example configuration that should generate your command (or come close):

```
[config]

PROCESS_LIST = PCPCombine

LOOP_BY = VALID
VALID_TIME_FMT = %Y%m%d%H
VALID_BEG = 2021032500
VALID_END = 2021032500
VALID_INCREMENT = 6H
LEAD_SEQ = 24

FCST_PCP_COMBINE_RUN = True
FCST_PCP_COMBINE_METHOD = USER_DEFINED

FCST_PCP_COMBINE_INPUT_TEMPLATE = {init?fmt=%Y%m%d}T0000Z_um_diag.nc
FCST_PCP_COMBINE_COMMAND = -add {FCST_PCP_COMBINE_INPUT_TEMPLATE} 'name="precipitation_amount"; level="(0,*,*)";' {FCST_PCP_COMBINE_INPUT_TEMPLATE} 'name="precipitation_amount"; level="(1,*,*)";' {FCST_PCP_COMBINE_INPUT_TEMPLATE} 'name="precipitation_amount"; level="(2,*,*)";' {FCST_PCP_COMBINE_INPUT_TEMPLATE} 'name="precipitation_amount"; level="(3,*,*)";'

FCST_PCP_COMBINE_OUTPUT_DIR = {OUTPUT_BASE}/pcp_combine
FCST_PCP_COMBINE_OUTPUT_TEMPLATE = {init?fmt=%Y%m%d}_sample_fcst{lead?fmt=%H}.nc
```

Let me know if this approach accomplishes what you need.
